### ISSUE TITLE
engine: dns interference detection functionality

### ISSUE THREAD

**Issue by c2xusnpq6** [ooni/probe#1497]
Can you add a `dns interference level test`?

Because in China, DNS pollution is very serious. It is not necessarily done by the government. The network operator on their side is also very complicated. The network operator will mess with the customer's connection.

(google translated...😅)

**Comment by c2xusnpq6** [comment]
https://help.nextdns.io/t/x2hmvas/what-is-dns-over-tls-dot-dns-over-quic-doq-and-dns-over-https-doh-doh3

**Comment by c2xusnpq6** [comment]
https://adguard.com/en/adguard-dns/overview.html

```
Adguard DOQ:
quic://dns.adguard.com

Adguard DOH:
https://dns.adguard.com/dns-query
```

**Comment by c2xusnpq6** [comment]
Cloudflare also supports two request modes, allowing for more detailed dns interference testing

**Comment by hellais** [comment]
Thanks for the feedback! We have an experimental test which investigates specifically the blocking of DoH, DoT and DoH3 servers. It's called dnscheck: https://github.com/ooni/spec/blob/master/nettests/ts-028-dnscheck.md.

We are still sorting out the last details about how we are going to actually deploy it in production, but you input on other DNS providers to look into is super helpful.

**Comment by c2xusnpq6** [comment]
It would be better if the user can test the dns server he wants to test.

**Comment by bassosimone** [comment]
Yes, I very much agree with you. I think we need to do two things: make Web Connectivity more capable of doing automatic things for automated runs and adding more support of running custom scans of DNS, HTTP, etc.

**Comment by c2xusnpq6** [comment]
```
https://1.1.1.1/dns-query
https://8.8.8.8/resolve
https://8.8.8.8/dns-query
https://101.101.101.101/dns-query
```

**Comment by c2xusnpq6** [comment]
my friends from mainland China told me that some network operators, communities, and local governments will block `1.1.1.1` and |8.8.8.8`

try the backup server, `1.0.0.1`, `8.8.4.4` or something like that

**Comment by c2xusnpq6** [comment]
I heard from people in mainland China that `alidns` will return blank or wrong ip value to users, such as google search and youtube related domain names


I can't confirm this, I can't confirm this, I tried it for a while and it worked...
So it may be treated separately according to the request source ip

**Comment by hellais** [comment]
This has been shipped. Closing.
